.github/workflows/merge.yml

```diff
@@ -1265,6 +1270,24 @@ jobs:
         shell: bash
         if: always()
 
+      - name: Confirming connection errors
+        shell: bash
+        timeout-minutes: 1
+        run: |
+          incoming_connection_errors=$(rg "IncomingConnectionError" "${{ matrix.node_data_path }}" -c --stats | \
+            rg "(\d+) matches" | rg "\d+" -o) || { echo "Failed to find IncomingConnectionError error"; exit 0; }
+          if [ -z "$incoming_connection_errors" ]; then
+            echo "Doesn't find any IncomingConnectionError error !"
+          else
+            echo "Found $incoming_connection_errors IncomingConnectionError errors."
+          fi
+          if ! rg "UnexpectedEof" "${{ matrix.node_data_path }}" -c --stats; then
```

Comment:
W encounter other errors that are related to a connection being closed just because a previous connection to the same peer succeeded.

`UnexpectedEof`
`Custom { kind: Other, error: Right(Left(Right(Select(Failed)))) }`
`error: Io(Os { code: 104, kind: ConnectionReset, message: \"Connection reset by peer\" }) })))))) } })`

Maybe it'd be hard to get info for all the types, we could just skip it and just log incomming connection errors.

Reply:
the test `without ws` will have no such specific Eof error detected, hence I'd expect the same after the patch, when `ws` is enabled.

We'd expect a drop on the IncomingConnectionError as well.
which, I'd suggest to put a threshold in for the `with ws` test, to avoid any future regression

Reply:
Right makes sense!